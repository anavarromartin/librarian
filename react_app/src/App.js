import React, { Component } from 'react'
import OfficesPage from './offices/OfficesPage'
import OfficePage from './offices/OfficePage'
import { BrowserRouter, Route, Redirect } from "react-router-dom"
import { ToastContainer } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'
import AddBookPage from './books/AddBookPage'

class App extends Component {
    render() {
        return (
            <BrowserRouter>
                <div>
                    <Route path="/" exact component={OfficesPage} />
                    <Route path="/:officeName" exact strict component={OfficePage} />
                    <Route path="/:officeName/add-book" render={(props) => (
                        !!window.session ?
                            <AddBookPage {...props} />
                            :
                            null
                    )} />

                    <ToastContainer autoClose={3000} />
                </div>
            </BrowserRouter>
        )
    }
}

export default App
