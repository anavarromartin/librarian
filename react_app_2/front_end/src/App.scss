@import "global-styles";

.top-container {
  align-self: center;
  justify-self: center;
  display: grid;
  grid-template-rows: auto 1fr;
  grid-template-columns: 1fr;
  height: 100%;

}

.header--hidden {
  display: none;
  grid-row-start: 1;
}

.content {
  grid-row-start: 2;
  padding: 0 $content-padding;
}

@media (min-width: $breakpoint-tablet) {
  .header--hidden {
    display: grid;
  }

}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
