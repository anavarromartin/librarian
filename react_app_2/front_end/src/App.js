import React, {useState} from 'react';
import './App.scss';
import classNames from 'classnames'
import AppHeader from "./components/app-header/app-header"
import {BrowserRouter, Route, Switch} from "react-router-dom";
import ReturnBook from "./components/return-book/return-book"
import {borrowBook, getAvailableBooks, getCheckedOutBooks, getOfficeBooks, returnBook} from "./methods/book-methods";
import BorrowBook from "./components/borrow-book/borrow-book"
import LibraryBrowsing from "./components/library-browsing/library-browsing";

const renderComponent = (component, routeProps, renderProps) => {
    const allProps = {...routeProps, ...renderProps}
    return React.createElement(component, {...allProps}, null)
}

const RouteWithBackNav = props => {

    return (
        <Route
            path={props.path}
            exact
            render={routeProps => renderComponent(props.component, props, routeProps)}
        />
    )
}

const App = () => {
    const [backLocation, setBackLocation] = useState(null)
    const [headerVisible, setHeaderVisibility] = useState(true)
    const [headerConfig, setHeaderConfig] = useState({})

    const navigateBack = (history) => {
        console.log('back location:' + backLocation)
        history.push(backLocation)
    }

    const customSetBackLocation = (loc) => {
        console.log(`setting location: ${loc}`)
        setBackLocation(loc)
    }

    const componentProps = {
        setBackLocation: setBackLocation,
        setHeaderVisibility: setHeaderVisibility,
        getCheckedOutBooks: getCheckedOutBooks,
        getAvailableBooks: getAvailableBooks,
        returnBook: returnBook,
        borrowBook: borrowBook,
        getOfficeBooks: getOfficeBooks,
        setHeaderConfig: setHeaderConfig
    }


    const backButtonEnabled = () => backLocation !== null

    return (
        <BrowserRouter basename={process.env.PUBLIC_URL}>
            <div className={"top-container"}>
                <div className={classNames({"header--hidden": !headerVisible})}>
                    <Route
                        path={"/:officeId"}
                        render={props => <AppHeader {...props}
                                                    onNavigateBack={navigateBack}
                                                    headerConfig={headerConfig}
                                                    backButtonEnabled={backButtonEnabled}
                        />}
                    />
                </div>
                <div className={"content"}>
                    <Switch>
                        <RouteWithBackNav path={`/:officeId/return`} component={ReturnBook} {...componentProps}/>
                        <RouteWithBackNav path={`/:officeId/borrow`} component={BorrowBook} {...componentProps}/>
                        <RouteWithBackNav path={`/:officeId`} component={LibraryBrowsing} {...componentProps}/>
                    </Switch>
                </div>
            </div>
        </BrowserRouter>
    )
}


export default App;
