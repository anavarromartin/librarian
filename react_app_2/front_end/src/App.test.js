import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import {shallow} from 'enzyme'

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App/>, div);
    ReactDOM.unmountComponentAtNode(div);
});

it('matches the snapshot', () => {
    expect(shallow(<App/>)).toMatchSnapshot()
})
