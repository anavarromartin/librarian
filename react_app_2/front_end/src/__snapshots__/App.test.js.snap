// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<BrowserRouter
  basename=""
>
  <div
    className="top-container"
  >
    <div
      className=""
    >
      <Route
        path="/:officeId"
        render={[Function]}
      />
    </div>
    <div
      className="content"
    >
      <Switch>
        <RouteWithBackNav
          borrowBook={[Function]}
          component={[Function]}
          getAvailableBooks={[Function]}
          getCheckedOutBooks={[Function]}
          getOfficeBooks={[Function]}
          path="/:officeId/return"
          returnBook={[Function]}
          setBackLocation={[Function]}
          setHeaderConfig={[Function]}
          setHeaderVisibility={[Function]}
        />
        <RouteWithBackNav
          borrowBook={[Function]}
          component={[Function]}
          getAvailableBooks={[Function]}
          getCheckedOutBooks={[Function]}
          getOfficeBooks={[Function]}
          path="/:officeId/borrow"
          returnBook={[Function]}
          setBackLocation={[Function]}
          setHeaderConfig={[Function]}
          setHeaderVisibility={[Function]}
        />
        <RouteWithBackNav
          borrowBook={[Function]}
          component={[Function]}
          getAvailableBooks={[Function]}
          getCheckedOutBooks={[Function]}
          getOfficeBooks={[Function]}
          path="/:officeId"
          returnBook={[Function]}
          setBackLocation={[Function]}
          setHeaderConfig={[Function]}
          setHeaderVisibility={[Function]}
        />
      </Switch>
    </div>
  </div>
</BrowserRouter>
`;
